### DISCUSSION TITLE
[expect-type] usage with create-react-app

### DISCUSSION THREAD

**Discussion by LorenzHenk** [mmkal/ts#141] [Q&A] [answered]
I'd like to use this library.

to test it, I've created a new react-app with typescript (`yarn create react-app myapp --template typescript`).

In the existing `App.test.tsx`, I've added the following statement to the existing test:
```
  expectTypeOf(1).toBeString();
```

In the IDE (I'm using VS Code), this shows an error (`An argument for 'MISMATCH' was not provided.`).
But when running the test with `yarn test`, no errors are shown.
Test files don't seem to be type checked.

How did you enable type checking for your test files?

**Comment by mmkal** [comment]
I'm not familiar with create-react-app - is there a script that runs the typescript compiler? That's where the errors would he caught.

**Comment by LorenzHenk** [comment]
Seems like the TS config is embedded here: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js#L102

**Comment by mmkal** [comment]
Is there a `yarn build` command or similar that catches type errors? That's what it suggests in [the create-react-app docs](https://create-react-app.dev/docs/adding-typescript/): 

> Type errors will show up in the same console as the build one. You'll have to fix these type errors before you continue development or build your project

**Comment by mmkal** [comment]
@LorenzHenk have converted this to a Q&A Discussion since I don't think there's an issue with expect-type here. Did you get it working with create-react-app in the end?

**Comment by LorenzHenk** [comment, marked as answer]
@mmkal you're right, `yarn build` fails.
Running the command `tsc` fails as well.

I just added the script
```
"ts:check": "tsc"
```
to `package.json` and can now run `yarn test && yarn ts:check` to make sure the tests are working correctly.
